### PR TITLE
Reliability: fix issues

### PR DESCRIPTION
**Pull request by juliuspf**
fixes https://github.com/jasp-stats/jasp-test-release/issues/1035
fixes https://github.com/jasp-stats/jasp-test-release/issues/1036
should fix the very large object size from issue (https://github.com/jasp-stats/jasp-issues/issues/999), needs to be verified still
The R-package `Bayesrel` should be updated to 0.6.5.6 (sorry for the constant updates)

### CODE REVIEW

**Comment by boutinb** [comment]
@vandenman Can you review this?

File: inst/qml/ReliabilityBayesian.qml [hunk 1]
```diff
@@ -272,32 +272,35 @@ Form
 			
 			IntegerField
 			{
+                id:             noSamples
```
Comment (vandenman) [inline inst/qml/ReliabilityBayesian.qml:275]: Tabs in QML
```suggestion
				id:             noSamples
```

File: R/reliabilityAnalysisFrequentist.R [hunk 2]
```diff
@@ -439,7 +452,7 @@ reliabilityFrequentist <- function(jaspResults, dataset, options) {
         allData <- cbind(allData, newData)
         
         # produce footnote for coefficients that are prone to fail with singular matrices, such as lambda6 and omega
-        if (is.nan(model[["cfi"]][["scaleCfi"]][[i]])) 
+        if (any(is.nan(model[["cfi"]][["scaleCfi"]][[i]])))
```
Comment (vandenman) [inline R/reliabilityAnalysisFrequentist.R:455]: if you're using `is.nan` here because `NA` is fine but `NaN` is not, it's fine. Otherwise I'd suggest:
```suggestion
        if (anyNA(model[["cfi"]][["scaleCfi"]][[i]]))
```

**Review by vandenman — CHANGES_REQUESTED** [review]
Two small comments below, to test this in JASP I have to copy all the files to jasp-desktop so that will take a little longer.

File: R/reliabilityAnalysisFrequentist.R (hunk 2 above)
Comment (juliuspf) [reply R/reliabilityAnalysisFrequentist.R:455]: If I remember correctly, this was specifically for the NaNs

File: inst/qml/ReliabilityBayesian.qml (hunk 1 above)
Comment (juliuspf) [reply inst/qml/ReliabilityBayesian.qml:275]: do you do the qml files with Qt?

**Comment by vandenman** [comment]
I'm still getting this:

image

with the latest master branch

File: R/reliabilityAnalysisFrequentist.R (hunk 2 above)
Comment (vandenman) [reply R/reliabilityAnalysisFrequentist.R:455]: then this is fine.

File: R/reliabilityAnalysisFrequentist.R [hunk 3]
```diff
@@ -422,7 +435,7 @@ reliabilityFrequentist <- function(jaspResults, dataset, options) {
   idxSelected <- which(selected)
   
   if (options[["mcDonaldScale"]] && !is.null(relyFit[["freq"]][["omega.error"]])) {
-    model[["footnote"]] <- gettextf("%sMcDonald's \u03C9 estimation method switched to PFA because the CFA
+    model[["footnote"]] <- gettextf("%sMcDonald's \u03C9 estimation method switched to PFA because the CFA
```
Comment (vandenman) [inline R/reliabilityAnalysisFrequentist.R:438]: @juliuspf can you undo this change?

File: R/reliabilityAnalysisFrequentist.R [hunk 4]
```diff
@@ -422,8 +435,7 @@ reliabilityFrequentist <- function(jaspResults, dataset, options) {
   idxSelected <- which(selected)
   
   if (options[["mcDonaldScale"]] && !is.null(relyFit[["freq"]][["omega.error"]])) {
-    model[["footnote"]] <- gettextf("%sMcDonald's \u03C9 estimation method switched to PFA because the CFA
-                                    did not find a solution. ", model[["footnote"]])
+    model[["footnote"]] <- gettextf("%sMcDonald's %s estimation method switched to PFA because the CFA did not find a solution. ", model[["footnote"]], "\u03C9")
```
Comment (vandenman) [inline R/reliabilityAnalysisFrequentist.R:438]: @juliuspf there's still a change here

Comment (juliuspf) [reply R/reliabilityAnalysisFrequentist.R:438]: I thought this was a change that Frans forgot, cause he did change the gettexts with omega in the rest of the files in the same way

**Review by vandenman — APPROVED** [review]

File: R/reliabilityAnalysisFrequentist.R (hunk 4 above)
Comment (vandenman) [reply R/reliabilityAnalysisFrequentist.R]: yeah probably. But I don't want there to be any differences between this file and the one in jasp-desktop, otherwise, the translations files also need to be changed again.

File: R/reliabilityAnalysisBayesian.R [hunk 5]
```diff
@@ -175,30 +175,28 @@ reliabilityBayesian <- function(jaspResults, dataset, options) {
...
       # add the scale info
-      corsamp <- apply(relyFit[["Bayes"]][["covsamp"]], c(1, 2), .cov2cor.callback, progressbarTick)
-      relyFit[["Bayes"]][["samp"]][["avg_cor"]] <- coda::mcmc(apply(corsamp, c(2, 3), function(x) mean(x[x!=1])))
+      relyFit[["Bayes"]][["samp"]][["avg_cor"]] <- matrix(0, dim(relyFit[["Bayes"]][["covsamp"]])[1], dim(relyFit[["Bayes"]][["covsamp"]])[2])
+      for (i in 1:dim(relyFit[["Bayes"]][["covsamp"]])[1]) {
+        for (j in 1:dim(relyFit[["Bayes"]][["covsamp"]])[2]) {
+          corm <- .cov2cor.callback(relyFit[["Bayes"]][["covsamp"]][i, j, , ], progressbarTick)
+          relyFit[["Bayes"]][["samp"]][["avg_cor"]][i, j] <- mean(corm[corm!=1])
+        }
+      }
```
Comment (vandenman) [inline R/reliabilityAnalysisBayesian.R:184]: Using `nrow` and `ncol` is slightly more readable, (`nrow(x) = dim(x)[1L]`, likewise for `ncol(x)`).

Considering `mean(corm[corm!=1])`, what's the goal here? Compute the average correlation without the diagonal? If yes, this is not an ideal way of doing that because (1) limits of floating point comparison and (2) you're counting every value double (which doesn't change the result but is unnecessary). How about:

```suggestion
      lowerTriangleIndex = which(lower.tri(relyFit[["Bayes"]][["covsamp"]][i, j, , ]))
      for (i in 1:nrow(relyFit[["Bayes"]][["covsamp"]])) {
        for (j in 1:ncol(relyFit[["Bayes"]][["covsamp"]])) {
          corm <- .cov2cor.callback(relyFit[["Bayes"]][["covsamp"]][i, j, , ], progressbarTick)
          relyFit[["Bayes"]][["samp"]][["avg_cor"]][i, j] <- mean(corm[lowerTriangleIndex])
        }
      }
```
you could adapt `.cov2cor.callback` to handle lower triangular input as well but I don't think that's worth the effort.

File: R/reliabilityAnalysisBayesian.R [hunk 6]
```diff
@@ -212,13 +210,20 @@ reliabilityBayesian <- function(jaspResults, dataset, options) {
...
+      relyFit[["n.item"]] <- p
+      
+      # free some memory
+      relyFit[["Bayes"]][["covsamp"]] <- NULL
+      relyFit[["data"]] <- NULL
+      relyFit[["Bayes"]][["data_mis_samp_cov"]] <- NULL
+      relyFit[["Bayes"]][["data_mis_samp_fm"]] <- NULL
+      
+      
       # Consider stripping some of the contents of relyFit to reduce memory load
```
Comment (vandenman) [inline R/reliabilityAnalysisBayesian.R:227]: delete this comment in light of what you added above?

File: R/reliabilityAnalysisBayesian.R [hunk 7]
```diff
@@ -275,12 +280,20 @@ reliabilityBayesian <- function(jaspResults, dataset, options) {
...
   for (nm in names(samps)) {
-    if (anyNA(samps[[nm]])) {
+    
+    if (length(dim(samps[[nm]])) == 3) {
+      samp_tmp <- apply(samps[[nm]], 3, as.vector)
+    } else {
+      samp_tmp <- as.vector(samps[[nm]])
+    }
+    
+    if (anyNA(samp_tmp)) {
```
Comment (vandenman) [inline R/reliabilityAnalysisBayesian.R:290]: shouldn't the `anyNA` check preceed the creation of `samp_tmp`?

File: R/reliabilityAnalysisBayesian.R [hunk 8]
```diff
@@ -575,7 +588,7 @@ reliabilityBayesian <- function(jaspResults, dataset, options) {
 
 
   if (!is.null(relyFit)) {
-    n.item <- dim(relyFit[["Bayes"]][["covsamp"]])[3]
+    n.item <- relyFit[["n.item"]]
```
Comment (vandenman) [inline R/reliabilityAnalysisBayesian.R:591]: good idea, much more readable!

**Review by vandenman — CHANGES_REQUESTED** [review]
looks good, some small comments below. I like how many changes improve the readability of the code!

**Review by vandenman — APPROVED** [review]

**Comment by vandenman** [comment]
@juliuspf do you still want to make more changes or is this about it?

**Comment by juliuspf** [comment]
I think that is it, anything else will go into the new structure
